admin/cluster_containers_setup.py

```diff
@@ -0,0 +1,324 @@
+            NOTE: alternative of making it work with env variables pending
+            implementation
+        :param options: ``ContainerOptions`` with the options passed to the
+            the script.
+        """
+        # XXX add the capability to use env variables if the options are not
+        # passed??
+        # self.control_node_address = env['FLOCKER_ACCEPTANCE_CONTROL_NODE']
+        # self.certificates_path = FilePath(
+        #    env['FLOCKER_ACCEPTANCE_API_CERTIFICATES_PATH'])
+        self.options = options
+        try:
+            self.application_template = self.options['template']
+            self.per_node = self.options['apps-per-node']
+            self.control_node_address = self.options['control-node']
+            self.timeout = self.options['wait']
+            if self.timeout is None:
+                self.timeout = 100
+        except Exception as e:
+            sys.stderr.write("%s: %s\n" % ("Missing or wrong arguments", e))
+            sys.stderr.write(e.args[0])
+            sys.stderr.write('\n\n')
+            sys.stderr.write(options.getSynopsis())
+            sys.stderr.write('\n')
+            sys.stderr.write(options.getUsage())
+            raise SystemExit(1)
+
+        certificates_path = FilePath(self.options['cert-directory'])
+        self.cluster_cert = certificates_path.child(b"cluster.crt")
+        self.user_cert = certificates_path.child(b"user.crt")
+        self.user_key = certificates_path.child(b"user.key")
+        self.client = None
+        self.reactor = reactor
+        self.nodes = []
+        self._initialise_client()
+
+    def _initialise_client(self):
+        """
+        Initialise flocker client.
+        """
+        self.client = FlockerClient(
+            self.reactor,
+            self.control_node_address,
+            REST_API_PORT,
+            self.cluster_cert,
+            self.user_cert,
+            self.user_key
+        )
+
+    def _set_nodes(self, nodes):
+        """
+        Set the list of the nodes in the cluster.
+        """
+        self.nodes = nodes
+
+    def deploy(self):
+        """
+        Deploy the new configuration: create the requested containers
+        and dataset in the cluster nodes.
+
+        :return Deferred: that will fire once the request to create all
+            the containers and datasets has been sent.
+        """
+        print "Listing current nodes"
+        d = self.client.list_nodes()
+        d.addCallback(self._set_nodes)
+        print "Building config"
+        d.addCallback(self._build_config)
+        print "Deploying new config"
+        d.addCallback(self._configure)
+        return d
+
+    def is_datasets_deployment_complete(self):
+        """
+        Check if all the dataset have been created.
+
+        :return `Deferred`: that will fire once the list datasets call
+            has been completed, and which result will bee True if all the
+            dataset have been created, or false otherwise.
```

Comment:
Remove backquotes from `Deferred`. Inside the `:return...:` clause, types do not need backquotes.  

Change the description to state simply that the Deferred "fires with True if all datasets have been created, or False otherwise".  You don't need to mention that the call has completed.  That is an implementation detail - if the code changed to use magic to determine whether all datasets have been created, then the simpler description would still be true - the current longer description would be wrong.